src/plots/grid.js

```diff
@@ -0,0 +1,409 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../lib');
+var domainAttrs = require('./domain').attributes;
+var counterRegex = require('../lib/regex').counter;
+var cartesianIdRegex = require('./cartesian/constants').idRegex;
+
+
+var gridAttrs = exports.attributes = {
+    rows: {
+        valType: 'integer',
+        min: 1,
+        role: 'info',
+        editType: 'calc',
```

Comment:
Good call - I used `calc` because that's what we had for cartesian `axis.domain` but neither it nor the grid attributes actually needed it -> https://github.com/plotly/plotly.js/pull/2399/commits/6b8d461f05eb5daee02caf9f13d2a7619303b6f5